File: longest_palindromic_substring/longestPalindromicSubstring.js
const longestPalindrome = (s) => {
	answer = s.substring(0,1);
	for (let i = 0; i < s.length; i++) {
		temp = expand(s, i, i);
		if (temp.length > answer.length) { answer = temp }
		temp = expand(s,i, i+1);
		if (temp.length > answer.length) { answer = temp }
	}
	return answer;
};

const expand = (s, first, last) => {
	while(s[first] === s[last] && first >= 0 && last < s.length) {
		first -= 1
		last += 1
	}
	return s.substring(first+1, last)
}

File: longest_palindromic_substring/longest_palindromic_substring.py
class Solution:
    def longestPalindrome(self, s):
        answer = s[0:1]
        for index in range(len(s)):
            temp = self.expand(s, index, index)
            if len(temp) > len(answer):
                answer = temp
            temp = self.expand(s, index, index+1)
            if len(temp) > len(answer):
                answer = temp
        return answer

    def expand(self, s, first, last):
        while first >= 0 and last < len(s) and s[first] == s[last]:
            first -= 1
            last += 1
        return s[first+1:last]
